examples/nginx-grafana-metabase/README.md

```diff
@@ -0,0 +1,162 @@
+
+## The Shape of Nginx's Access Log Data
+In the [previous article](https://www.timeplus.com/post/analyzing-nginx-access-logs), we briefly touched on the Nginx access logs but did not dwell on its location or how the data it contains should be parsed.
+
+### Nginx Access Log Data
+The access logs for Nginx are typically written to the file `/var/log/nginx/access.log`. Here's an excerpt from my blog's access logs:
+```bash
+161.35.230.x - - [26/Jun/2023:06:33:53 +0000] "\x00\x0E8uON\x85J\xCF\xC5\x93\x00\x00\x00\x00\x00" 400 182 "-" "-"
+51.79.29.xx - - [26/Jun/2023:06:37:04 +0000] "POST / HTTP/1.1" 301 57 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36"
+179.43.177.xxx - - [26/Jun/2023:06:45:28 +0000] "GET / HTTP/1.1" 301 60 "-" "Hello World"
+87.236.176.xxx - - [26/Jun/2023:07:58:36 +0000] "GET / HTTP/1.1" 200 6364 "https://www.ayewo.com/" "Mozilla/5.0 (compatible; InternetMeasurement/1.0; +https://internet-measurement.com/)"
+87.236.176.xxx - - [26/Jun/2023:07:58:38 +0000] "GET /favicon.png HTTP/2.0" 200 7636 "-" "Mozilla/5.0 (compatible; InternetMeasurement/1.0; +https://internet-measurement.com/)"
+93.126.72.xxx - - [26/Jun/2023:08:29:38 +0000] "GET /how-to-install-rsync-on-windows/ HTTP/2.0" 200 6394 "https://www.google.com/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
+```
+
+The first line in the access log (from the IP address `161.35.230.x`) is actually a maliciously crafted request that doesn't even specify a HTTP method (i.e. `GET` or `OPTIONS`) to the server which is why the server responded with a HTTP 400 code (Bad Request). 
+
+There are several of such malformed requests in the access logs that are tricky to parse which is why an additional column `malicious_request` was added to the `nginx_historical_access_log` stream which was created in the previous article.
+
+### Nginx Access Log Format
+Nginx's [access log format](https://docs.nginx.com/nginx/admin-guide/monitoring/logging/#setting-up-the-access-log) is specified in `/etc/nginx/nginx.conf` and it broadly looks like this:
+```bash
+http {
+    log_format compression '$remote_addr - $remote_user [$time_local] '
+                           '"$request" $status $body_bytes_sent '
+                           '"$http_referer" "$http_user_agent"';
+
+    server {
+        access_log /var/log/nginx/access.log;
+        ...
+    }
+}
+```
+
+Below is a brief explanation of how each access log line is parsed into 12 columns:
+* `$remote_addr` is stored in column `remote_ip` of type `ipv4`;
+* `-` is stored in column `rfc1413_ident` of type `string`;
+* `$remote_user` is stored in column `remote_user` of type `string`;
+* `$time_local` os stored in column `date_time` of type `datetime64`;
+* `$request` is split into 4 columns each of type `string` to make our analysis easier:
```

Comment:
better to mention this string is separated by whitespace.  I would suggest omitting `;` in the column column, the user may get the wrong impression each part is separated by ; not white space. Also I suggest we talk about 4 parts in the requests, then mention how they are mapped to columns in Timeplus stream schema